### PR TITLE
[DOC] RDoc for Complex

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Could not get a link to local Kernel#Complex to work, so linked to docs.ruby-lang.org.

### CODE REVIEW

File: complex.c [hunk 1]
```diff
@@ -2350,30 +2364,27 @@ float_arg(VALUE self)
...
  *
- * == Rectangular Coordinates
+ * You can create a \Complex object from rectangular coordinates with:
  *
- * Each of the methods above (except Complex.polar) takes two "rectangular" arguments
- * representing the _real_ and _imaginary_ parts of the created \Complex object;
- * see {Complex definition}[https://en.wikipedia.org/wiki/Complex_number#Definition].
+ * - A {complex literal}[rdoc-ref:doc/syntax/literals.rdoc@Complex+Literals].
+ * - \Method Complex.rect.
+ * - \Method {Kernel#Complex}[https://docs.ruby-lang.org/en/master/Kernel.html#method-i-Complex],
```
Comment (peterzhu2118) [inline complex.c:2381]: What version of RDoc are you on? I'm on `6.6.1` and `Kernel#Complex` links correctly for me.

Comment (BurdetteLamar) [reply complex.c:2381]: $ rdoc --version
6.3.1

Comment (BurdetteLamar) [reply complex.c]: I know that Kernel#Complex links correctly on docs.ruby-lang.org, so have complied here.

File: complex.c [hunk 2]
```diff
@@ -523,39 +523,53 @@ static VALUE nucomp_s_convert(int argc, VALUE *argv, VALUE klass);
...
- *   digit = "0" | "1" | "2" | "3" | "4" | "5" | "6" | "7" | "8" | "9" ;
- *   extra spaces = ? \s* ? ;
- *
- * See String#to_c.
+ *   Complex(abs, arg = 0, exception: true) -> complex or nil
+ *   Complex(s, exception: true) -> complex or nil
+ *
+ * Returns a new \Complex object if the arguments are valid;
+ * otherwise raises an exception if +exception+ is +true+;
+ * otherwise returns +nil.
```
Comment (peterzhu2118) [inline complex.c:531]:
```suggestion
 * otherwise returns +nil+.
```

File: complex.c [hunk 3]
```diff
@@ -523,39 +523,53 @@ static VALUE nucomp_s_convert(int argc, VALUE *argv, VALUE klass);
...
+ *
+ * With Numeric argument +abs+, returns Complex.rect(abs, arg) if the arguments are valid.
```
Comment (peterzhu2118) [inline complex.c:533]:
```suggestion
 * With Numeric argument +abs+, returns <tt>Complex.rect(abs, arg)</tt> if the arguments are valid.
```

**Review by peterzhu2118 — APPROVED** [review]

File: complex.c [hunk 4]
```diff
@@ -523,39 +523,53 @@ static VALUE nucomp_s_convert(int argc, VALUE *argv, VALUE klass);
...
- *
- * See String#to_c.
+ *   Complex(abs, arg = 0, exception: true) -> complex or nil
+ *   Complex(s, exception: true) -> complex or nil
+ *
+ * Returns a new \Complex object if the arguments are valid;
+ * otherwise raises an exception if +exception+ is +true+;
+ * otherwise returns +nil+.
+ *
+ * With Numeric argument +abs+, returns <tt>Complex.rect(abs, arg)</tt> if the arguments are valid.
```
Comment (nobu) [inline complex.c:533]: Ditto.

**Pull request merged**